source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,237 @@
+===========
+Search Text
+===========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you will learn how to perform a text search in MongoDB.
+
+Performing a text search allows you to search a string field in your
+collection for terms or phrases. This is useful for searching
+unstructured text such as transcripts, resumes, essays, or web pages. 
+
+Sample Documents
+~~~~~~~~~~~~~~~~
+
+In the following examples, a junior year student is browsing through
+computer science courses to enroll in at their college. The documents in
+the ``junior_courses`` collection represents the courses available for
+the student to select from: 
+
+.. code-block:: json
+
+    { "_id": 1, "course": "Artificial Intelligence" }
+    { "_id": 2, "course": "Discrete Mathematical Structures for Computer Science" }
+    { "_id": 3, "course": "Web Database Applications" }
+    { "_id": 4, "course": "Computer Science Seminar" }
+    { "_id": 5, "course": "Computer Hacking Revealed" }
+    { "_id": 6, "course": "Advanced Computer Game Programming" }
+    { "_id": 7, "course": "Image Processing in Computer Science" }
+
+Text Index
+----------
+
+Before you preform a text search, you must specify the field to search
+in a text index for your collection. 
+
+The text index specifies the fields to query during a text search. There can
+only be one text index per collection with a field whose value is a
+string or an array of string elements. 
+
+In the following examples, we perform text searches on the ``course``
+field in the ``junior_courses`` collection. To enable text searches on
+the ``course`` field, create a text index using the following snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+If there was another string field, ``description``, you wanted to index,
+you can create a compound index to perform a text search in both fields
+using the following snippet: 
+
+.. code-block:: java
+
+    collection.createIndex(Indexes.compoundIndex(Indexes.text("course"), Indexes.text("description")));
+
+For more information, see our server manual entry on :manual:`text
+indexes </core/index-text/>`.
+
+Searching
+---------
+
+To perform a text search, use the ``text()`` method of the ``Filters``
+builder. 
+
+The ``text()`` method performs a text search on the fields indexed with
+a text index. The text search matches documents which contain the
+specified search term. 
+
+You can include ``TextSearchOptions`` as the second parameter to specify
+additional text search options to handle case sensitivity, diacritic
```

Comment:
I think case sensitivity is probably a common use case. I think it would be great if you could mention whether searches are case sensitive by default. And it would be wonderful to have an example in which you're setting it to whatever is not the default.